.github/workflows/run-kola-tests.yaml

```
@@ -60,53 +60,163 @@ jobs:
+          name: ${{ matrix.arch }}-binpkgs
 
-      - name: Run tests
+      - name: Download test update image from other workflow
+        uses: gabriel-samfira/action-download-artifact@v5
+        if: ${{ inputs.workflow_run_id }}
+        with:
+          workflow: ${{ inputs.workflow_name_or_id }}
+          workflow_conclusion: success
+          run_id: ${{ inputs.workflow_run_id }}
+          name: ${{ matrix.arch }}-test-update
+
+      - name: Download generic image from other workflow
+        uses: gabriel-samfira/action-download-artifact@v5
+        if: ${{ inputs.workflow_run_id }}
+        with:
+          workflow: ${{ inputs.workflow_name_or_id }}
+          workflow_conclusion: success
+          run_id: ${{ inputs.workflow_run_id }}
+          name: ${{ matrix.arch }}-generic-image
+
+      - name: Download developer container from other workflow
+        uses: gabriel-samfira/action-download-artifact@v5
+        if: ${{ inputs.workflow_run_id }}
+        with:
+          workflow: ${{ inputs.workflow_name_or_id }}
+          workflow_conclusion: success
+          run_id: ${{ inputs.workflow_run_id }}
+          name: ${{ matrix.arch }}-devcontainer
+
+      - name: Download torcx tarball from other workflow
+        uses: gabriel-samfira/action-download-artifact@v5
+        if: ${{ inputs.workflow_run_id }}
+        with:
+          workflow: ${{ inputs.workflow_name_or_id }}
+          workflow_conclusion: success
+          run_id: ${{ inputs.workflow_run_id }}
+          name: ${{ matrix.arch }}-torcx
+
+      - name: Extract artifacts
         shell: bash
         run: |
           exec 2>&1
-          set +x
+          set -x
           set -euo pipefail
 
-          # extract the image.
-          IMG_ARCHIVE=$(readlink -f images/**/flatcar_production_image.bin.bz2)
+          DEVCONTAINER_WEBROOT="scripts/devcontainer-webroot"
+          echo "DEVCONTAINER_WEBROOT=${DEVCONTAINER_WEBROOT}" >> "$GITHUB_ENV"
+
+          # Set up webroot for devcontainer and torcx tests.
+          # The respective tests will download devcontainer and torcx tarball via http.
+          # The devcontainer test will then run a build
+          #   which will download and install binpkgs into the dev container.
+          # For the sake of that test we will serve both via a temporary local web server.
+          mkdir ${DEVCONTAINER_WEBROOT}
+          mv flatcar_developer_container* ${DEVCONTAINER_WEBROOT}
+          tar -C ${DEVCONTAINER_WEBROOT} -xcf binpkgs.tar.gz
+
+          tar -C $[DEVCONTAINER_WEBROOT} -xf torcx.tar
+
+          echo "------------"
+          ls -laR ${DEVCONTAINER_WEBROOT}
+          echo "------------"
+
+          # Extract the generic image we'll use for qemu tests.
+          # Note that the qemu[_uefi] tests use the generic image instead of the
+          #  qemu vendor VM image ("Astronaut: [...] Always have been.").
+          IMG_ARCHIVE=$(readlddink -f images/**/flatcar_production_image.bin.bz2)
```

```suggestion
          IMG_ARCHIVE=$(readlink -f images/**/flatcar_production_image.bin.bz2)
```
Or is this something else?

---

That's a leftover which isn't actually needed anymore (no subdirectories anymore for image artifacts). Will remove, please pardon the confusion.